### PR TITLE
feat: add durationUpdate callback

### PR DESCRIPTION
**Pull request by BuddyLReno**
### Platforms affected
- Android
- iOS

### What does this PR do?
- Add support for a durationUpdate callback.

This particular change helps deal with the fact that Android shuts down webapps in the background after a few minutes. You can subscribe to this callback and let the audio player report the duration to you when it's available instead of querying for it.

### What testing has been done on this change?
Physical device testing on:

- iPhone X (iOS 12)
- iPhone 6 (iOS 11)
- OnePlus 5T (Oreo 8.1)
- Samsung Galaxy S8 (Nougat 7.1)

### Checklist
- [ ] [Reported an issue](http://cordova.apache.org/contribute/issues.html) in the JIRA database
- [ ] Commit message follows the format: "CB-3232: (android) Fix bug with resolving file paths", where CB-xxxx is the JIRA ID & "android" is the platform affected.
- [ ] Added automated test coverage as appropriate for this change.

### CODE REVIEW

**Comment by BuddyLReno** [comment]
This particular change helps deal with the fact that Android shuts down webapps in the background after a few minutes. You can subscribe to this callback and let the audio player report the duration to you when it's available instead of querying for it.

**Comment by BuddyLReno** [comment]
@janpio updated! Sorry about that!

File: www/Media.js [hunk 1]
```diff
@@ -35,16 +35,22 @@ var mediaObjects = {};
  * @param errorCallback         The callback to be called if there is an error.
  *                                  errorCallback(int errorCode) - OPTIONAL
  * @param statusCallback        The callback to be called when media status has changed.
- *                                  statusCallback(int statusCode) - OPTIONAL
```
Comment (janpio) [inline www/Media.js]: This line should probably not be moved down, right?

**Comment by janpio** [comment]
(Thanks for the updates on the text, I cleaned up the comments a bit by hiding.)

Interesting change. I understand both parts of what you wrote:

> This particular change helps deal with the fact that Android shuts down webapps in the background after a few minutes.

> You can subscribe to this callback and let the audio player report the duration to you when it's available instead of querying for it.

But to be honest I don't really get how those are connected. Can you elaborate maybe?

File: www/Media.js (hunk 1 above)
Comment (BuddyLReno) [reply www/Media.js]: ah that was a mistake!

Comment (BuddyLReno) [reply www/Media.js]: Fixed!

**Comment by BuddyLReno** [comment]
Yes, I can elaborate.

In the newer versions of android, after about five minutes of background activity, android shuts down the web container that is running the app. So when you try to use a background task to query for the duration with a `setTimeout` as is described in the docs, that setTimeout never actually runs.  This occurs even if you use a plugin like cordova-plugin-background-mode.

What I mean by subscribing, is basically just attaching a callback to allow the native code to run the callback when the audio player assigns duration. By doing this, you always get a real duration and don't have to worry about getting a "-1" duration as it only reports the duration after the audio player actually initializes and has the real duration. This removes the need to query for duration with a setTimeout, which prevents the situation above from happening. 

Does that help?

**Comment by BuddyLReno** [comment]
Great! I'm getting ready to leave the office but I'll update the README later tonight after the kids go to bed. 👍

**Comment by BuddyLReno** [comment]
@janpio ugh, i'm sorry this took until Monday to get to! My weekend was a lot busier than I had planned on. I've updated the README with the Android quirk requested! :)

File: README.md [hunk 2]
```diff
@@ -237,6 +239,9 @@ var timerDur = setInterval(function() {
 }, 100);
 ```
 
+#### Android Quirk
+Newer versions of android have aggressive routines that limit background processing. If you are trying to get duration while your app is in the background longer than roughly 5 minutes, you will get more consistent results by using the `mediaDurationUpdate` callback.
```
[the remark this answers is not in the document]
Comment (BuddyLReno) [reply README.md]: Love it! ❤️

**Comment by BuddyLReno** [comment]
@janpio would it be alright if I created a test app using Ionic? I'm most familiar with that platform.

**Comment by janpio** [comment]
Sure, whatever you are comfortable with. Shouldn't be to hard to recreate with Cordova though later on. Maybe I can fork your app then and transform it to pure Cordova.

**Comment by BuddyLReno** [comment]
@janpio here's the demo project! Let me know if you have issues getting it to install. https://github.com/BuddyLReno/cordova-plugin-media-duration-demo

**Comment by BuddyLReno** [comment]
Once we get this pr merged in, I'm going to update the media dependency to use the public one.

**Review by breautek — APPROVED** [review]

**Review by erisu — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -235,6 +237,9 @@ var timerDur = setInterval(function() {
 }, 100);
 ```
 
+#### Android Quirk
```
Comment (erisu) [inline README.md:240]:
```suggestion
#### Android Quirk

```
Just adding a new line between the header and content for MD consistency.

**Pull request merged**